cli/compose/interpolation/interpolation.go

```diff
@@ -42,7 +42,7 @@ func interpolateSectionItem(
 		if err != nil {
 			return nil, errors.Errorf(
 				"Invalid interpolation format for %#v option in %s %#v: %#v. You may need to escape any $ with another $.",
-				key, section, name, err.Template,
+				key, section, name, err.(*template.InvalidTemplateError).Template,
```

Comment:
Ya, maybe to be saw we can do this instead of  `if err != nil`

```
switch err := err.(type)
case nil:
case template.InvalidTemplateError:
    return nil, errors.Errorf(...)
default:
    return nil, errors.Wrapf(err, "error while interpolating %s in %s %s", key, section, name)


```